### PR TITLE
Fix `h`'s coverage in the tests themselves

### PR DESCRIPTION
**Pull request by jon-betts**
This doesn't get us to the end, but at least now if you see coverage gaps in the tests, you know you added them.

A nice next step would be to get the number of files with test coverage issues down to the point where they fit on one screen.

### CODE REVIEW

File: tests/h/streamer/messages_test.py [hunk 1]
```diff
@@ -38,8 +38,8 @@ def test_it_handles_a_full_queue(self, _handler, work_queue):
 
         _handler({"foo": "bar"})
 
-        result = work_queue.get_nowait()
-        assert result.topic == "queue_is_full"
+        result = work_queue.get_nowait()  # pragma: nocover
+        assert result.topic == "queue_is_full"  # pragma: nocover
```
Comment (jon-betts) [inline tests/h/streamer/messages_test.py:42]: I don't get this one. It's clearly covered

File: tests/h/activity/bucketing_test.py [hunk 2]
```diff
@@ -24,9 +24,9 @@ def __eq__(self, timeframe):
             and self.document_buckets == timeframe.document_buckets
         )
 
-    def __repr__(self):
-        # pragma: no cover
-        return f'{self.__class__} "{self.label}" with {len(self.document_buckets)} document buckets'
+    # pragma: nocover
+    def __repr__(self):  # pragma: nocover
+        return f'{self.__class__} "{self.label}" with {len(self.document_buckets)} document buckets'  # pragma: nocover
```
Comment (jon-betts) [inline tests/h/activity/bucketing_test.py:29]: I've no idea which of these are doing what. I had real trouble getting this to stop being counted

File: tests/h/views/api/bulk/_ndjson_test.py [hunk 3]
```diff
@@ -31,11 +31,11 @@ def test_it_returns_204_if_no_content_is_to_be_returned(self):
...
     def test_it_captures_initial_errors(self):
-        def failing_method():
-            raise ValueError("Oh no!")
+        def failing_method(fail=True):
+            if fail:
+                raise ValueError("Oh no!")
 
-            # pragma: nocover
-            yield 1  # pylint: disable=unreachable
+            yield 1  # pragma: nocover
```
Comment (jon-betts) [inline tests/h/views/api/bulk/_ndjson_test.py:38]: There was some fight between unreachable and coverage. So I made it "reachable"

File: tests/h/activity/query_test.py [hunk 4]
```diff
@@ -404,12 +404,8 @@ def test_it_returns_each_annotations_group(self, _fetch_groups, pyramid_request)
...
                 presented_annotations.extend(bucket.presented_annotations)
 
-        for group in _fetch_groups.return_value:
-            for presented_annotation in presented_annotations:
-                if presented_annotation["group"] == group:
-                    break
-            else:
-                assert False
+        expected_groups = [anno["group"] for anno in presented_annotations]
+        assert _fetch_groups.return_value == expected_groups
```
Comment (jon-betts) [inline tests/h/activity/query_test.py:408]: This is both simpler, and also much more obvious about what is being asserted. At a larger scale I've no idea what this test file is up to.

File: tests/h/search/query_test.py [hunk 5]
```diff
@@ -1,4 +1,3 @@
-# pylint: disable=too-many-lines
```
Comment (jon-betts) [inline tests/h/search/query_test.py:1]: The refactors here took this file under 1000 lines. Still some work to do in the brevity department.

File: tests/h/activity/bucketing_test.py (hunk 2 above)
Comment (marcospri) [reply tests/h/activity/bucketing_test.py:29]: Needs more no cover

**Review by marcospri — APPROVED** [review]
Looks good, removing that unused fixture is very satisfying.

**Pull request merged**